test/semantics/se-cgen-sql-new.test.js

```diff
@@ -0,0 +1,175 @@
+const { api, pipe } = require('../../src/rhyme')
+const { rh, parse } = require('../../src/parser')
+const { compile } = require('../../src/simple-eval')
+const { preproc } = require('../../src/preprocess')
+const { typing, types } = require('../../src/typing')
+
+const fs = require('node:fs/promises')
+const os = require('node:child_process')
+
+let execPromise = function (cmd) {
+  return new Promise(function (resolve, reject) {
+    os.exec(cmd, function (err, stdout) {
+      if (err) return reject(err);
+      resolve(stdout);
+    });
+  });
+}
+
+test("testTrivial", async () => {
+  let query = rh`1 + 200`
+
+  let func = compile(query, { backend: "c-sql-new", schema: types.nothing })
+
+  let res = await func()
+  expect(res).toEqual("res = 201\n")
+})
```

Comment:
Is "res = ..." part of the expected output? That would seems a little strange, I think we'd expect just "201" here

Reply:
I can change that. I have the result printed like that because I wanted to distinguish between the result of the query and the output from the stateful "print" because the result of print is currently an i8 which is just set to zero. If we have a column with 5, 2, 1, 7, in stdout it will look like:
```
5
2
1
7
0
```
where the last 0 is the result of the query and not part of the column.

I think we can make the result of print be types.nothing so that if the query is a print, there will not be meaningful output  and we just want the side effects (printf's). In the generator I can choose to not print the result if the res has type nothing.